crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -79,8 +79,23 @@ pub(crate) fn fd_fdstat_get(fd: &File) -> Result<wasi::__wasi_fdflags_t> {
     Ok(fdflags)
 }
 
-pub(crate) fn fd_fdstat_set_flags(fd: &File, fdflags: wasi::__wasi_fdflags_t) -> Result<()> {
-    unimplemented!("fd_fdstat_set_flags")
+pub(crate) fn fd_fdstat_set_flags(fd: &mut File, fdflags: wasi::__wasi_fdflags_t) -> Result<()> {
+    let handle = unsafe { fd.as_raw_handle() };
+
+    let access_mode = winx::file::query_access_information(handle)?;
+
+    let access_mode = file_access_mode_from_fdflags(
+        fdflags,
+        (access_mode & AccessMode::FILE_READ_DATA).bits() != 0,
+        (access_mode & (AccessMode::FILE_WRITE_DATA | AccessMode::FILE_APPEND_DATA)).bits() != 0,
+    );
+
+    let new_handle =
+        winx::file::reopen_file(handle, access_mode, file_flags_from_fdflags(fdflags))?;
+
+    *fd = unsafe { File::from_raw_handle(new_handle) };
```

Comment:
This will only work if `fd` points to a correct place inside the underlying data structures used in `wasi_ctx`. This should be documented, at the very least. Besides, this won't work for stdout/stdin, which have no underlying `File`.

Reply:
> This will only work if `fd` points to a correct place inside the underlying data structures used in `wasi_ctx`. This should be documented, at the very least. Besides, this won't work for stdout/stdin, which have no underlying `File`.

Hmm, stdout/in are still just ordinary handles on Windows, so I don't see why this shouldn't work for them. Could you explain why you think that this is the case?

Reply:
I mean that on all platforms we don't store a `File` in the `Descriptor` if it's stdin/stderr/stdout and create a fresh handle from `io::stdout()` on every access instead. Therefore, we'll overwrite some instace of file, but it'll be thrown away pretty soon. We probably should replace the whole `descriptor` field or the whole `FdEntry`.

Btw. we probably should have at least smoke tests for `fd_fdstat_set_flags` with stdout/stdin/stderr.

Reply:
> I mean that on all platforms we don't store a `File` in the `Descriptor` if it's stdin/stderr/stdout and create a fresh handle from `io::stdout()` on every access instead. Therefore, we'll overwrite some instace of file, but it'll be thrown away pretty soon. We probably should replace the whole `descriptor` field or the whole `FdEntry`.

That's a good observation.

> 
> Btw. we probably should have at least smoke tests for `fd_fdstat_set_flags` with stdout/stdin/stderr.

Sounds like a good idea but personally I'd suggest we do that in a subsequent PR.

Reply:
This has been changed to return a new `OsHandle` as needed which the platform-agnostic implementation respects to update the existing descriptor.